packages/harpguru-core/src/components/option-stack/utils/are-option-stacks-equal/are-option-stacks-equal.ts

```diff
@@ -34,45 +34,21 @@ from ${prevPropsz.length} to ${nextPropsz.length}
     `)
   }
 
-  // It's important to return this early before the iterative
-  // tests which happen later are performed so that we can
-  // don't have to confuse that logic with protective statements
-  // in case the arrays aren't the same length
-  //
-  // (this is relevant *once* we have more granular tests of
-  // the items included here)
-  const areListLengthsEqual = prevPropsz.every(
-    ({ items: prevItems }, index) =>
-      prevItems.length === nextPropsz[index].items.length
-  )
-  if (!areListLengthsEqual) {
-    if (!isProd()) return false
-
-    throw Error(`
-The memoised OptionStack components should be initialised with
-parameters which never go stale. Please check the params provided
-to this OptionStack.
-
-One of the option lists has changed length during this render.
-    `)
-  }
-
+  // TODO: change all of these checks to `some` to prevent unnecessary
+  // completion of the iteration.
   const areTitlesEqual = prevPropsz.every(
-    (prevProps, index) => prevProps.title === nextPropsz[index].title
-  )
-  const areUseSubTitlesEqual = prevPropsz.every(
-    (prevProps, index) =>
-      prevProps.useSubTitle === nextPropsz[index].useSubTitle
-  )
-  const areItemTapHandlersEqual = prevPropsz.every(
-    (prevProps, index) =>
-      prevProps.itemTapHandler === nextPropsz[index].itemTapHandler
+    (prevProps, index) => prevProps.useTitle === nextPropsz[index].useTitle
   )
+  // TODO: Add check that the useItems functions are identical
```

Comment:
TODO here. Make sure that it all still works in production mode after this change too. Don't want to realise 10 commits later.

Reply:
Done, and checking in production mode highlighted an issue with the way the items for the scales menu were being produced.